src/utils.jl

```diff
@@ -94,6 +94,8 @@ Base.setindex!(D::ColVecs, v::AbstractVector, i) = setindex!(D.X, v, :, i)
 
 Base.vcat(a::ColVecs, b::ColVecs) = ColVecs(hcat(a.X, b.X))
 Base.zero(x::ColVecs) = ColVecs(zero(x.X))
+Base.reduce(::typeof(hcat), a::ColVecs) = a.X
```

Comment:
I wonder if we should return
```suggestion
Base.reduce(::typeof(hcat), a::ColVecs) = copy(a.X)
```
The result of `reduce(hcat, ...)` and `reduce(vcat, ...)` generally, and currently for `RowVecs` and `ColVecs`, does not share the same memory as the input collection. Maybe it's too unexpected and potentially dangerous to share memory here.

Reply:
this is a really good question. I tried to find documentation whether `hcat` and `vcat` is allowed to share memory and I could not find any resource. Not having a `!` might hint at it being copy, but on the other hand it does not modify the contents and `reshape` also shares memory. And I would place `hcat`, `vcat` into the same category. But this is truly an underspecification of `hcat` `vcat`. I'll try to find a good answer to this

Reply:
> but on the other hand it does not modify the contents and reshape also shares memory.

Yes, it would apply to that case as well. BTW it's not always desirable that a function returns a reshaped view, see, e.g., https://github.com/JuliaLang/julia/issues/33143 and https://github.com/JuliaLang/julia/issues/43507.

Reply:
I pushed a `copy` version because I am starting to get the notion that this is the desired behavior. Unfortunately this eats most of the performance gains for `ColVecs`